### ISSUE TITLE
autocert / letsencrypt integration

### ISSUE THREAD

**Issue by fortytw2** [bitly/oauth2_proxy#313]
Would be extremely useful for oauth2_proxy to automatically grab SSL certificates from LetsEncrypt if asked to. I'm willing to implement this, if the maintainers think it's a good / mergable idea :)

**Comment by jehiah** [comment]
@fortytw2 yes! I'd be happy to see this land using `https://github.com/rsc/letsencrypt`

**Comment by fortytw2** [comment]
@jehiah fantastic! I'll try to send a PR tomorrow :)
